src/whitenoise/groups.rs

```diff
@@ -862,6 +874,84 @@ impl Whitenoise {
         })
     }
 
+    /// Downloads and decrypts a chat media blob
+    ///
+    /// This helper orchestrates the download from Blossom, hash verification,
+    /// and MDK decryption in a single focused operation.
+    ///
+    /// # Arguments
+    /// * `account_pubkey` - The account downloading the media
+    /// * `data_dir` - The data directory for MDK storage
+    /// * `group_id` - The MLS group ID
+    /// * `media_file` - The MediaFile record containing URLs, hashes, and metadata
+    /// * `original_file_hash` - SHA-256 of original content (for MDK decryption)
+    ///
+    /// # Returns
+    /// * `Ok(Vec<u8>)` - Decrypted file data
+    /// * `Err(WhitenoiseError)` - If download, verification, or decryption fails
+    async fn download_and_decrypt_chat_media_blob(
+        account_pubkey: &PublicKey,
+        data_dir: &Path,
+        group_id: &GroupId,
+        media_file: &MediaFile,
+        original_file_hash: &[u8; 32],
+    ) -> Result<Vec<u8>> {
+        // Extract filename required for MDK AAD (cryptographically bound)
+        let filename = media_file
+            .file_metadata
+            .as_ref()
+            .and_then(|m| m.original_filename.as_deref())
+            .ok_or_else(|| {
+                WhitenoiseError::MediaCache("Missing required filename metadata".to_string())
+            })?;
+        // Parse encrypted_file_hash for download verification
```

Comment:
I don't fully understand for what the encrypted_file_hash is necessary... couldn't this be checked after decryption? comparing with the original file hash? And that way we can save only one hash?